library/core/src/main/java/com/google/android/exoplayer2/util/Util.java

```diff
@@ -850,6 +855,17 @@ public static int inferContentType(String fileName) {
   }
 
   /**
+   * Returns the whether the device can output 32bit float audio
+   *
+   * @param capabilities The audio capabilities of the device.
+   * @param channelCount The channel count.
+   */
+  public static boolean canHandle32BitFloatAudio(AudioCapabilities capabilities, int channelCount) {
+    return (capabilities != null && capabilities.supportsEncoding(C.ENCODING_PCM_FLOAT)
+     && ((Util.SDK_INT > 20 && channelCount <= 2) || (Util.SDK_INT >= 24)));
```

Comment:
I have no idea.  I only have a Shield hooked up to a receiver and it's up to date.  I would have never thought to put my nexus players (running 7.1 now) in my theater.  I'll take a gander through change logs on this later.  Wouldn't surprise me if it was a shield only limitation at the earlier apis.